lib/src/rules/directives_ordering.dart

```diff
@@ -229,6 +302,31 @@ class _Visitor extends SimpleAstVisitor {
         .forEach(reportDirective);
   }
 
+  void _checkSectionInOrder(
+      Set<AstNode> lintedNodes, Iterable<NamespaceDirective> nodes) {
+    void reportDirective(NamespaceDirective directive) {
+      if (lintedNodes.add(directive)) {
+        rule._reportLintWithDirectiveSectionOrderedAlphabeticallyMessage(
+            directive);
+      }
+    }
+
+    List<NamespaceDirective> addIfUpperBound(
+        List<NamespaceDirective> acc, NamespaceDirective curr) {
+      if (acc.isEmpty || acc.last.uriContent.compareTo(curr.uriContent) < 0) {
+        acc.add(curr);
+      }
+
+      return acc;
+    }
+
+    final inOrderNodes = nodes.fold(<NamespaceDirective>[], addIfUpperBound);
+
+    nodes.toSet()
+      ..removeAll(inOrderNodes)
+      ..forEach(reportDirective);
```

Comment:
In that case, the last code would mark d2, and it is more intuitive to mark d5 because it is the one you have to move, it could be done changing the code for this:

```
NamespaceDirective previousDirective = null;
for (NamespaceDirective directive in nodes) {
  if (previousDirective != null && previousDirective.uriContent.compareTo(directive.uriContent) > 0) {
    reportDirective(previousDirective);
  }
  previousDirective = directive;
}
```
but in the case you have d7,d1,d3,d4,d2,d5,d6  the second code would lint d7 and d4, d7 is OK because you have to move it, but d4 not, you should move d2 to fix that, so I think a good way to reduce lints and give a better hint is mark both, with this code.

```
NamespaceDirective previousDirective = null;
for (NamespaceDirective directive in nodes) {
  if (previousDirective != null && previousDirective.uriContent.compareTo(directive.uriContent) > 0) {
    reportDirective(directive);
    reportDirective(previousDirective);
  }
  previousDirective = directive;
}
```
In that case the user would see this:

import 'd7.dart'; // LINT
import 'd1.dart'; // LINT
import 'd3';
import 'd4.dart'; // LINT
import 'd2.dart'; // LINT
import 'd5';
import 'd6';

I think that is a pretty good solution,other one could be  calculate the nodes you have to move with minimal cost of sortering a block moving directives from its line to another then mark them, but it is more sophisticated and hard to implement, what do you think?